src/EnergyPlus/ChillerAbsorption.cc

```diff
@@ -109,19 +106,18 @@ const char *moduleObjectType("Chiller:Absorption");
 const char *fluidNameWater = "WATER";
 const char *fluidNameSteam = "STEAM";
 
-PlantComponent *BLASTAbsorberSpecs::factory(EnergyPlusData &state, std::string const &objectName)
+BLASTAbsorberSpecs *BLASTAbsorberSpecs::factory(EnergyPlusData &state, std::string const &objectName)
 {
     // Process the input data
     if (state.dataChillerAbsorber->getInput) {
         GetBLASTAbsorberInput(state);
         state.dataChillerAbsorber->getInput = false;
     }
     // Now look for this particular object
-    for (auto &thisAbs : state.dataChillerAbsorber->absorptionChillers) {
-        if (thisAbs.Name == objectName) {
-            return &thisAbs;
-        }
-    }
+    auto thisAbs = std::find_if(state.dataChillerAbsorber->absorptionChillers.begin(),
```

Comment:
Now as a future step, if we define `std::string name` on the PlantComponent base class, we could make a base class function to search by the `name` variable.  This could replace all the individual lambda find-ifs.  Obviously these are not a big deal, but just saying one way it could go.